src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,110 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setAge(18);
+        user.setLogin("UserLogin");
+        user.setPassword("qwerty");
+    }
+
+    @Test
+    void register_userIsValid_ok() {
+        User actual = registrationService.register(user);
+        assertEquals(user,actual);
+    }
+
+    @Test
+    void register_userIsNull_notOk() {
+        assertThrows(RuntimeException.class, () ->
+                registrationService.register(null));
+    }
+
+    @Test
+    void register_loginIsNull_notOk() {
+        user.setLogin(null);
+        assertThrows(RuntimeException.class, () ->
+                registrationService.register(user));
+    }
+
+    @Test
+    void register_withExistLogin_notOk() {
+        User user1 = new User();
```

Comment:
```suggestion
        User firstUser = new User();
```
using numbers in variable names is considered bad practice